master/buildbot/status/build.py

```diff
@@ -629,6 +629,14 @@ def get_artifacts(self):
                 self.artifacts = artifacts
             return self.artifacts
 
+    def get_trigger_steps_time(self):
+        trigger = 0
+        trigger_steps = [step.finished - step.started for step in self.steps if step.finished and str(TriggerType) == step.step_type ]
```

Comment:
Are you ok with the approach of checking step type? Would it be nicer to have it built in step status or make it configurable? Just a few other thoughts, say if we get to exclude other steps(hypothetically not derived from Trigger), or include this one back, then each time source code has to be patched.